### PR TITLE
Automatically install the probe-rs binaries on request

### PR DESCRIPTION
**Pull request by Yatekii**

### CODE REVIEW

File: src/domain.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+async function findExecutable(executableName: string): Promise<string | null> {
+    const envPath = process.env.PATH || '';
+    const envExt = process.env.PATHEXT || '';
+    const pathDirs = envPath.replace(/["]+/g, '').split(path.delimiter).filter(Boolean);
+    const extensions = envExt.split(';');
+    const candidates = pathDirs.flatMap((d) =>
+        extensions.map((ext) => path.join(d, executableName + ext)),
+    );
+    try {
+        console.log('CHECKING');
```
Comment (bugadani) [inline src/domain.ts]: Useful and informative log mesage.

File: src/domain.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+async function checkFileExists(filePath): Promise<string | null> {
+    if ((await fs.stat(filePath)).isFile()) {
+        return filePath;
+    }
+    return null;
+}
+
+export async function probeRsInstalled() {
+    if (await findExecutable('probe-rs')) {
```
Comment (bugadani) [inline src/domain.ts]: Why isn't this `return (await ...`?

File: src/extension.ts [hunk 3]
```diff
@@ -536,6 +549,74 @@ function startDebugServer(
...
+            };
+
+            const exitListener = (code: number | null, signal: NodeJS.Signals | null) => {
+                let message;
+                if (code === 0) {
+                    message = 'Installation successful.';
+                } else if (signal) {
+                    message = 'Installation aborted.';
+                } else {
+                    message = 'Installation failed.';
```
Comment (bugadani) [inline src/extension.ts]: Should we add some help, like "go to probe.rs for setup instructions"?

**Comment by bugadani** [comment]
Okay one more thing from me, though that's a bit of a scope creep:

The plugin should read the binary's version, compare the major to the plugin's version and offer an update if the plugin is newer than the binary. While this algorithm isn't perfect (as it ignores patch updates), it would avoid incompatibility issues arising from a version mismatch.

**Comment by Yatekii** [comment]
This will come in a separate PR.

**Review by bugadani — APPROVED** [review]

**Comment by bugadani** [comment]
Awesomesauce :)

**Comment by noppej** [comment]
@Yatekii This is going to be awesome for new users. Are you planning on updating the installation instructions also?

**Comment by Yatekii** [comment]
Hmm why is this not merging 🤔

**Pull request merged**